### PR TITLE
Common: allow overriding params

### PR DESCRIPTION
**Pull request by jochem-brouwer**
Closes https://github.com/ethereumjs/ethereumjs-monorepo/issues/3505

This PR expands the types in Common such that one can override params in hardforks.

An example is given in the 2935 test on how to use it.

### CODE REVIEW

File: packages/common/src/types.ts [hunk 1]
```diff
@@ -196,6 +201,22 @@ export type HardforkConfig = {
...
+  pow?: {
+    [key: string]: number | bigint | null
+  }
+  sharding?: {
+    [key: string]: number | bigint | null
+  }
+  vm?: {
+    [key: string]: number | bigint | null
+  }
 } & EIPOrHFConfig
```
Comment (holgerd77) [inline packages/common/src/types.ts:220]: I think you don't need to do this, this is already in through `&EIPOrHFConfig`

Comment (jochem-brouwer) [reply packages/common/src/types.ts:220]: Hey, you are right, I think this got introduced in #3500

**Review by holgerd77 — COMMENTED** [review]
Can we maybe have a test case for the `Common` part? Or do we already have one?

**Comment by jochem-brouwer** [comment]
I will add a test case!

**Review by holgerd77 — APPROVED** [review]
Cool, thanks, LGTM!

**Pull request merged**